torch-clone.sh

```diff
@@ -17,3 +18,6 @@ done
 # torchbenchmark needs to have this name and be in the same folder as
 # PyTorch, otherwise benchmarks/dynamo/torchbench.py won't find it
 git clone git@github.com:pytorch/benchmark.git "torchbenchmark"
+
+cd pytorch
+pip install -r requirements.txt -r .ci/docker/requirements-ci.txt
```

Comment:
Why is this necessary? If anything, this should go in the conda env.

Reply:
Well, there is at least one library in requirements-ci.txt which isn't in `pytorch-dev.yaml`.

I note that these requirements seem to stay fairly constant, but surely it would better not to duplicate them in two places?

For the moment, I'll remove it, and if I find any missing libraries in a new environment that cause issues, I'll add them to `pytorch-dev.yaml`.